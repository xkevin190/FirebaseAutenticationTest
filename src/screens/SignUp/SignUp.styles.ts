import {StyleSheet, ImageStyle, ViewStyle} from 'react-native';

import {COLORS} from '../../constants/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.GRAY_LIGHT,
    alignItems: 'center',
  } as ViewStyle,
  inputsContainer: {
    width: '100%',
    paddingHorizontal: 24,
    marginBottom: 8,
  } as ViewStyle,

  signInButton: {
    marginTop: 32,
  } as ViewStyle,
});

export default styles;
