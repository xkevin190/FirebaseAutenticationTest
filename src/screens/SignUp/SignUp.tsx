import React, {useState, useCallback} from 'react';
import {
  View,
  TouchableOpacity,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
  Keyboard,
} from 'react-native';
import {useTranslation} from 'react-i18next';
import {useFocusEffect, useNavigation} from '@react-navigation/native';
import Icon from 'react-native-vector-icons/Ionicons';

import type {useNavigationRootStack} from '../../types/navigation';
import {LOGIN} from '../../navigation/routes';
import {CustomInput, Button, HyperText} from '../../shared';
import {COLORS} from '../../constants/styles';
import CreateAccountCompleted from './components/CreateAccountCompleted';
import styles from './SignUp.styles';
import useAppDispatch from '../../hooks/useAppDispatch';
import {useSingUpForm} from './hooks/useSignUpForm';
import Controller from '../../shared/Controller';
import {createAccountThunk, authActions} from '../../store/Auth/slice';
import {CreateAccountRequest, User} from '../../types/FirebaseService';
import useAppSelector from '../../hooks/useAppSelector';
import {accountCreated, authLoading} from '../../store/Auth/selectors';

const ControlledInputs = Controller(CustomInput);

const SignUp: React.FC = () => {
  const [isPasswordHidden, setIsPasswordHidden] = useState(true);
  const {t} = useTranslation();
  const navigation = useNavigation<useNavigationRootStack>();
  const dispatch = useAppDispatch();
  const successful = useAppSelector(accountCreated);
  const loading = useAppSelector(authLoading);

  const {control, watch, handleSubmit} = useSingUpForm({
    email: '',
    lastName: '',
    firstName: '',
    password: '',
  });

  const createAccount = (data: CreateAccountRequest) => {
    dispatch(
      createAccountThunk({
        email: data.email,
        password: data.password,
        firstName: data.firstName,
        lastName: data.lastName,
      }),
    );
  };

  useFocusEffect(
    React.useCallback(() => {
      return () => {
        dispatch(authActions.resetSucess());
      };
    }, []),
  );

  return !successful ? (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
      <View style={styles.container}>
        <KeyboardAvoidingView style={styles.inputsContainer} behavior="padding">
          <ControlledInputs
            name="firstName"
            control={control}
            title={t('signUp:translation.nameInput.label')}
            placeholder={t('signUp:translation.nameInput.placeholder')}
            value={watch('firstName')}
          />

          <ControlledInputs
            name="lastName"
            control={control}
            value={watch('lastName')}
            title={t('signUp:translation.lastNameInput.label')}
            placeholder={t('signUp:translation.lastNameInput.placeholder')}
          />

          <ControlledInputs
            name="email"
            control={control}
            value={watch('email')}
            title={t('signUp:translation.emailInput.label')}
            placeholder={t('signUp:translation.emailInput.placeholder')}
          />

          <ControlledInputs
            name="password"
            control={control}
            value={watch('password')}
            title={t('signUp:translation.passwordInput.label')}
            placeholder={t('signUp:translation.passwordInput.placeholder')}
            password={isPasswordHidden}
            icon={
              <TouchableOpacity
                onPress={() => setIsPasswordHidden(!isPasswordHidden)}>
                <Icon
                  name={isPasswordHidden ? 'eye' : 'eye-off'}
                  size={15}
                  color={COLORS.MAIN_DEFAULT}
                />
              </TouchableOpacity>
            }
          />
        </KeyboardAvoidingView>
        <Button
          buttonText={t(
            `signUp:translation.createAccountButton.${
              loading ? 'loading' : 'label'
            }`,
          )}
          textColor={COLORS.WHITE}
          rounded
          testID="createAccountButton"
          disabled={loading}
          onPress={handleSubmit(createAccount)}
        />
        <HyperText
          prefix={t('signUp:translation.signIn.prefix')}
          text={t('signUp:translation.signIn.label')}
          customStyles={styles.signInButton}
          onPress={() => navigation.navigate(LOGIN)}
          testID={'signInButton'}
        />
      </View>
    </TouchableWithoutFeedback>
  ) : (
    <CreateAccountCompleted />
  );
};

export default SignUp;
