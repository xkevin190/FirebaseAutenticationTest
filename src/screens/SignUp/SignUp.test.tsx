import React from 'react';
import {
  render,
  fireEvent,
  RenderResult,
  waitFor,
} from '@testing-library/react-native';
import {NavigationContext} from '@react-navigation/native';
import SignUp from './SignUp';

let mockDispatch = jest.fn();
jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');

jest.mock('../../hooks/useAppDispatch', () => {
  return () => mockDispatch;
});

jest.mock('../../hooks/useAppSelector', () => jest.fn());
jest.mock('../../store/Auth/selectors', () => jest.fn());

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

jest.mock('../../store/Auth/slice', () => {
  return {
    createAccountThunk: jest.fn(),
    authActions: {
      resetError: jest.fn(),
      resetSucess: jest.fn(),
      clearErrors: jest.fn(),
    },
  };
});

let Component: RenderResult;

beforeEach(() => {
  Component = render(
    <NavigationContext.Provider value={navContext}>
      <SignUp />
    </NavigationContext.Provider>,
  );
});

const navContext = {
  isFocused: () => true,

  addListener: jest.fn(() => jest.fn()),
};

describe('SignUp component', () => {
  test('renders all input fields', () => {
    const {getByPlaceholderText, toJSON} = Component;
    const firstNameInput = getByPlaceholderText(
      'signUp:translation.nameInput.placeholder',
    );
    const lastNameInput = getByPlaceholderText(
      'signUp:translation.lastNameInput.placeholder',
    );
    const emailInput = getByPlaceholderText(
      'signUp:translation.emailInput.placeholder',
    );
    const passwordInput = getByPlaceholderText(
      'signUp:translation.passwordInput.placeholder',
    );

    expect(toJSON()).toMatchSnapshot();
    expect(firstNameInput).toBeTruthy();
    expect(lastNameInput).toBeTruthy();
    expect(emailInput).toBeTruthy();
    expect(passwordInput).toBeTruthy();
  });

  test('submits the form with valid values', async () => {
    const {getByPlaceholderText, getByTestId} = Component;
    const firstNameInput = getByPlaceholderText(
      'signUp:translation.nameInput.placeholder',
    );
    const lastNameInput = getByPlaceholderText(
      'signUp:translation.lastNameInput.placeholder',
    );
    const emailInput = getByPlaceholderText(
      'signUp:translation.emailInput.placeholder',
    );
    const passwordInput = getByPlaceholderText(
      'signUp:translation.passwordInput.placeholder',
    );
    const submitButton = getByTestId('createAccountButton');

    fireEvent.changeText(firstNameInput, 'John');
    fireEvent.changeText(lastNameInput, 'Doe');
    fireEvent.changeText(emailInput, 'test@gmail.com');
    fireEvent.changeText(passwordInput, 'K123456v?');

    expect(firstNameInput.props.value).toBe('John');
    expect(lastNameInput.props.value).toBe('Doe');
    expect(emailInput.props.value).toBe('test@gmail.com');
    expect(passwordInput.props.value).toBe('K123456v?');

    await waitFor(() => {
      fireEvent.press(submitButton);
    });
    expect(mockDispatch).toHaveBeenCalled();
  });

  test('displays error message for empty email field', async () => {
    const {getByPlaceholderText, debug, getByTestId, queryAllByTestId, update} =
      Component;
    const firstNameInput = getByPlaceholderText(
      'signUp:translation.nameInput.placeholder',
    );
    const lastNameInput = getByPlaceholderText(
      'signUp:translation.lastNameInput.placeholder',
    );
    const emailInput = getByPlaceholderText(
      'signUp:translation.emailInput.placeholder',
    );
    const passwordInput = getByPlaceholderText(
      'signUp:translation.passwordInput.placeholder',
    );
    const submitButton = getByTestId('createAccountButton');

    fireEvent.changeText(firstNameInput, '');
    fireEvent.changeText(lastNameInput, '');
    fireEvent.changeText(emailInput, '');
    fireEvent.changeText(passwordInput, '');
    fireEvent.press(submitButton);

    await waitFor(() => {
      expect(queryAllByTestId('helperText')).toHaveLength(4);
    });
  });
});
