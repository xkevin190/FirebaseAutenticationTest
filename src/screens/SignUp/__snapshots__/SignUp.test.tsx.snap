// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component renders all input fields 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F6F8FF",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "marginBottom": 8,
          "paddingHorizontal": 24,
          "width": "100%",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#9DA5B9",
          "fontFamily": "Lato-Bold",
          "fontSize": 12,
        }
      }
      testID=""
    >
      SIGNUP:TRANSLATION.NAMEINPUT.LABEL
    </Text>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#9DA5B9",
          "borderRadius": 16,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 18,
          "marginTop": 6,
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        control={
          {
            "_defaultValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_disableForm": [Function],
            "_executeSchema": [Function],
            "_fields": {
              "email": {
                "_f": {
                  "mount": true,
                  "name": "email",
                  "ref": {
                    "name": "email",
                  },
                  "value": "",
                },
              },
              "firstName": {
                "_f": {
                  "mount": true,
                  "name": "firstName",
                  "ref": {
                    "name": "firstName",
                  },
                  "value": "",
                },
              },
              "lastName": {
                "_f": {
                  "mount": true,
                  "name": "lastName",
                  "ref": {
                    "name": "lastName",
                  },
                  "value": "",
                },
              },
              "password": {
                "_f": {
                  "mount": true,
                  "name": "password",
                  "ref": {
                    "name": "password",
                  },
                  "value": "",
                },
              },
            },
            "_formState": {
              "dirtyFields": {},
              "disabled": false,
              "errors": {},
              "isDirty": false,
              "isLoading": false,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": {},
              "validatingFields": {},
            },
            "_formValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": {
              "array": Set {},
              "mount": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
              "unMount": Set {},
              "watch": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
            },
            "_options": {
              "defaultValues": {
                "email": "",
                "firstName": "",
                "lastName": "",
                "password": "",
              },
              "delayError": 100,
              "mode": "onBlur",
              "reValidateMode": "onChange",
              "resolver": [Function],
              "shouldFocusError": true,
            },
            "_proxyFormState": {
              "dirtyFields": false,
              "disabled": true,
              "errors": "all",
              "isDirty": false,
              "isValid": false,
              "isValidating": false,
              "touchedFields": false,
              "validatingFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_resetDefaultValues": [Function],
            "_setErrors": [Function],
            "_state": {
              "action": false,
              "mount": true,
              "watch": false,
            },
            "_subjects": {
              "array": {
                "next": [Function],
                "observers": [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "values": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateDisabledField": [Function],
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "handleSubmit": [Function],
            "register": [Function],
            "setError": [Function],
            "unregister": [Function],
          }
        }
        multiline={false}
        name="firstName"
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="signUp:translation.nameInput.placeholder"
        placeholderTextColor="#9DA5B9"
        secureTextEntry={false}
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
          }
        }
        textAlignVertical="auto"
        value=""
      />
    </View>
    <Text
      style={
        {
          "color": "#9DA5B9",
          "fontFamily": "Lato-Bold",
          "fontSize": 12,
        }
      }
      testID=""
    >
      SIGNUP:TRANSLATION.LASTNAMEINPUT.LABEL
    </Text>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#9DA5B9",
          "borderRadius": 16,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 18,
          "marginTop": 6,
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        control={
          {
            "_defaultValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_disableForm": [Function],
            "_executeSchema": [Function],
            "_fields": {
              "email": {
                "_f": {
                  "mount": true,
                  "name": "email",
                  "ref": {
                    "name": "email",
                  },
                  "value": "",
                },
              },
              "firstName": {
                "_f": {
                  "mount": true,
                  "name": "firstName",
                  "ref": {
                    "name": "firstName",
                  },
                  "value": "",
                },
              },
              "lastName": {
                "_f": {
                  "mount": true,
                  "name": "lastName",
                  "ref": {
                    "name": "lastName",
                  },
                  "value": "",
                },
              },
              "password": {
                "_f": {
                  "mount": true,
                  "name": "password",
                  "ref": {
                    "name": "password",
                  },
                  "value": "",
                },
              },
            },
            "_formState": {
              "dirtyFields": {},
              "disabled": false,
              "errors": {},
              "isDirty": false,
              "isLoading": false,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": {},
              "validatingFields": {},
            },
            "_formValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": {
              "array": Set {},
              "mount": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
              "unMount": Set {},
              "watch": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
            },
            "_options": {
              "defaultValues": {
                "email": "",
                "firstName": "",
                "lastName": "",
                "password": "",
              },
              "delayError": 100,
              "mode": "onBlur",
              "reValidateMode": "onChange",
              "resolver": [Function],
              "shouldFocusError": true,
            },
            "_proxyFormState": {
              "dirtyFields": false,
              "disabled": true,
              "errors": "all",
              "isDirty": false,
              "isValid": false,
              "isValidating": false,
              "touchedFields": false,
              "validatingFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_resetDefaultValues": [Function],
            "_setErrors": [Function],
            "_state": {
              "action": false,
              "mount": true,
              "watch": false,
            },
            "_subjects": {
              "array": {
                "next": [Function],
                "observers": [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "values": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateDisabledField": [Function],
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "handleSubmit": [Function],
            "register": [Function],
            "setError": [Function],
            "unregister": [Function],
          }
        }
        multiline={false}
        name="lastName"
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="signUp:translation.lastNameInput.placeholder"
        placeholderTextColor="#9DA5B9"
        secureTextEntry={false}
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
          }
        }
        textAlignVertical="auto"
        value=""
      />
    </View>
    <Text
      style={
        {
          "color": "#9DA5B9",
          "fontFamily": "Lato-Bold",
          "fontSize": 12,
        }
      }
      testID=""
    >
      SIGNUP:TRANSLATION.EMAILINPUT.LABEL
    </Text>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#9DA5B9",
          "borderRadius": 16,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 18,
          "marginTop": 6,
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        control={
          {
            "_defaultValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_disableForm": [Function],
            "_executeSchema": [Function],
            "_fields": {
              "email": {
                "_f": {
                  "mount": true,
                  "name": "email",
                  "ref": {
                    "name": "email",
                  },
                  "value": "",
                },
              },
              "firstName": {
                "_f": {
                  "mount": true,
                  "name": "firstName",
                  "ref": {
                    "name": "firstName",
                  },
                  "value": "",
                },
              },
              "lastName": {
                "_f": {
                  "mount": true,
                  "name": "lastName",
                  "ref": {
                    "name": "lastName",
                  },
                  "value": "",
                },
              },
              "password": {
                "_f": {
                  "mount": true,
                  "name": "password",
                  "ref": {
                    "name": "password",
                  },
                  "value": "",
                },
              },
            },
            "_formState": {
              "dirtyFields": {},
              "disabled": false,
              "errors": {},
              "isDirty": false,
              "isLoading": false,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": {},
              "validatingFields": {},
            },
            "_formValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": {
              "array": Set {},
              "mount": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
              "unMount": Set {},
              "watch": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
            },
            "_options": {
              "defaultValues": {
                "email": "",
                "firstName": "",
                "lastName": "",
                "password": "",
              },
              "delayError": 100,
              "mode": "onBlur",
              "reValidateMode": "onChange",
              "resolver": [Function],
              "shouldFocusError": true,
            },
            "_proxyFormState": {
              "dirtyFields": false,
              "disabled": true,
              "errors": "all",
              "isDirty": false,
              "isValid": false,
              "isValidating": false,
              "touchedFields": false,
              "validatingFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_resetDefaultValues": [Function],
            "_setErrors": [Function],
            "_state": {
              "action": false,
              "mount": true,
              "watch": false,
            },
            "_subjects": {
              "array": {
                "next": [Function],
                "observers": [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "values": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateDisabledField": [Function],
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "handleSubmit": [Function],
            "register": [Function],
            "setError": [Function],
            "unregister": [Function],
          }
        }
        multiline={false}
        name="email"
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="signUp:translation.emailInput.placeholder"
        placeholderTextColor="#9DA5B9"
        secureTextEntry={false}
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
          }
        }
        textAlignVertical="auto"
        value=""
      />
    </View>
    <Text
      style={
        {
          "color": "#9DA5B9",
          "fontFamily": "Lato-Bold",
          "fontSize": 12,
        }
      }
      testID=""
    >
      SIGNUP:TRANSLATION.PASSWORDINPUT.LABEL
    </Text>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#9DA5B9",
          "borderRadius": 16,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 18,
          "marginTop": 6,
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        control={
          {
            "_defaultValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_disableForm": [Function],
            "_executeSchema": [Function],
            "_fields": {
              "email": {
                "_f": {
                  "mount": true,
                  "name": "email",
                  "ref": {
                    "name": "email",
                  },
                  "value": "",
                },
              },
              "firstName": {
                "_f": {
                  "mount": true,
                  "name": "firstName",
                  "ref": {
                    "name": "firstName",
                  },
                  "value": "",
                },
              },
              "lastName": {
                "_f": {
                  "mount": true,
                  "name": "lastName",
                  "ref": {
                    "name": "lastName",
                  },
                  "value": "",
                },
              },
              "password": {
                "_f": {
                  "mount": true,
                  "name": "password",
                  "ref": {
                    "name": "password",
                  },
                  "value": "",
                },
              },
            },
            "_formState": {
              "dirtyFields": {},
              "disabled": false,
              "errors": {},
              "isDirty": false,
              "isLoading": false,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": {},
              "validatingFields": {},
            },
            "_formValues": {
              "email": "",
              "firstName": "",
              "lastName": "",
              "password": "",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": {
              "array": Set {},
              "mount": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
              "unMount": Set {},
              "watch": Set {
                "firstName",
                "lastName",
                "email",
                "password",
              },
            },
            "_options": {
              "defaultValues": {
                "email": "",
                "firstName": "",
                "lastName": "",
                "password": "",
              },
              "delayError": 100,
              "mode": "onBlur",
              "reValidateMode": "onChange",
              "resolver": [Function],
              "shouldFocusError": true,
            },
            "_proxyFormState": {
              "dirtyFields": false,
              "disabled": true,
              "errors": "all",
              "isDirty": false,
              "isValid": false,
              "isValidating": false,
              "touchedFields": false,
              "validatingFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_resetDefaultValues": [Function],
            "_setErrors": [Function],
            "_state": {
              "action": false,
              "mount": true,
              "watch": false,
            },
            "_subjects": {
              "array": {
                "next": [Function],
                "observers": [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "values": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateDisabledField": [Function],
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "handleSubmit": [Function],
            "register": [Function],
            "setError": [Function],
            "unregister": [Function],
          }
        }
        multiline={false}
        name="password"
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="signUp:translation.passwordInput.placeholder"
        placeholderTextColor="#9DA5B9"
        secureTextEntry={true}
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
          }
        }
        textAlignVertical="auto"
        value=""
      />
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "15%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Icon
            color="#313342"
            name="eye"
            size={15}
          />
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2162E3",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": "45%",
      }
    }
    testID="createAccountButton"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Lato-Regular",
          "fontSize": 14,
        }
      }
      testID=""
    >
      signUp:translation.createAccountButton.label
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginTop": 32,
        "opacity": 1,
      }
    }
    testID="signInButton"
  >
    <Text>
      <Text
        style={
          {
            "color": "#9DA5B9",
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
          }
        }
        testID=""
      >
        signUp:translation.signIn.prefix 
      </Text>
      <Text
        style={
          {
            "color": "#2162E3",
            "fontFamily": "Lato-Bold",
            "fontSize": 14,
          }
        }
        testID=""
      >
        signUp:translation.signIn.label
      </Text>
    </Text>
  </View>
</View>
`;
